shared/src/lib/utils.js

```diff
@@ -95,19 +97,41 @@ export async function summarizeContent({
   };
 }
 
-export async function updateSettings(handleGetData) {
+export async function fetchSettings() {
   const sessionObject = await browser.storage.local.get('session_token');
   const syncObject = await browser.storage.local.get('sync_existing');
   const apiObject = await browser.storage.local.get('api_token');
   const apiEngineObject = await browser.storage.local.get('api_engine');
 
-  await handleGetData({
+  return {
     token: sessionObject?.session_token,
     sync_existing:
       typeof syncObject?.sync_existing !== 'undefined'
         ? syncObject.sync_existing
         : true,
     api_token: apiObject?.api_token,
     api_engine: apiEngineObject?.api_engine,
+  };
+}
+
+export async function getActiveTab() {
+  const tabs = await browser.tabs.query({
+    active: true,
+    lastFocusedWindow: true,
   });
+
+  // Chrome/Firefox might give us more than one active tab when something like "chrome://*" or "about:*" is also open
+  const tab =
+    tabs.find(
+      (tab) =>
+        tab?.url?.startsWith('http://') || tab?.url?.startsWith('https://'),
+    ) || tabs[0];
+
+  if (!tab || !tab.url) {
```

Comment:
Found this new case, after permissions were granted, before they were requested, where we'd get a `tab` without a `url`. That would cause an "Invalid URL" error. The "fix" was simply to close the extension _after_ requesting the permissions for the first time (so we can load it up _with_ permissions).